docs/source/general/data.rst

```diff
@@ -2,11 +2,132 @@
 Data
 ####
 
+.. _data:
+
+*******************************
+Using DataModule + DataPipeline
+*******************************
+
+In this section, we will create a very simple ``ImageClassificationPreprocess`` with a ``ImageClassificationDataModule``.
+
+Example::
+
+    import os
+    import numpy as np
+    from flash.data.data_module import DataModule
+    from flash.data.process import Preprocess
+    from PIL.Image import Image
+    import torchvision.transforms as T
+    from torch import Tensor
+
+    # Subclass ``Preprocess``
+
+    class ImageClassificationPreprocess(Preprocess):
+
+        to_tensor = T.ToTensor()
+
+        def load_data(self, folder: str, dataset: AutoDataset) -> Iterable:
```

Comment:
This is auto-injected and optional. If users works with Preprocess, the AutoDataset is handled for them in behind the scenes.
However, it can useful to get access to it to store metadata.